module3-crud-and-data-models/r1-schema-and-data-models/README.md

```diff
@@ -0,0 +1,99 @@
+
+### CRUD and REST
+
+### What is REST?
+
+REST is a set of architectural constraints. API developers can implement REST in a variety of ways. When a client request is made via a RESTful API, it transfers a representation of the state of the resource to the requester or endpoint. This information, or representation, is delivered in one of several formats, JSON is the most generally popular file format to use.
+
+### CRUD (CREATE, READ, UPDATE, DELETE)
+
+In a REST environment, CRUD often corresponds to the HTTP methods `POST`, `GET`, `PUT`, and `DELETE`, respectively. These are the fundamental elements of a persistent storage system.
+
+For example, imagine we are working with a system that is keeping track of meals and their corresponding prices for a restaurant. Let’s look at how we would implement CRUD operations.
+
+#### Create
+
+To create resources in a REST environment, we most commonly use the HTTP POST method. POST creates a new resource of the specified resource type.
+Imagine that we are adding a new food item to the stored list of dishes for this restaurant, and the dish objects are stored in a dishes resource.
+
+Note: Use `POST` when you want to add a child resource under resources collection.
+
+##### Request:
+
+`POST http://www.myrestaurant.com/api/dishes/`
+
+We will need to send the `dish` data too.
+
+#### Read
+
+To read resources in a REST environment, we use the GET method. in practice, reading a resource shouldn't change any information - it should only retrieve it.
+
+To read resources in a REST environment, we use the GET method. Reading a resource should never change any information - it should only retrieve it. REST itself is more like a set of guidelines. Technically, you can change data in a `GET` request, but since we are creating a RESTful API, you shouldn't do that. Having a GET request that updates data in your database would be confusing to the users of your API and violate the expected behavior of REST.
+
+##### Request:
+
+`GET http://www.myrestaurant.com/api/dishes/`
+
+#### Update
+
+PUT is the HTTP method used for the CRUD operation, Update.
+So if the price of Avocado Toast has gone up, we should go into the database and update that information using PUT.
+
+Note: Use `PUT` when you want to modify a singular resource which is already a part of resources collection. `PUT` replaces the resource with the data you send in its entirety.
+
+##### Request:
+
+`PUT http://www.myrestaurant.com/dishes/:id`
+
+#### Delete
+
+The CRUD operation Delete corresponds to the HTTP method DELETE. It is used to remove a resource from the system.
+Let’s say that the world avocado shortage has reached a critical point, and we can no longer afford to serve this modern delicacy at all. We should go into the database and delete the item that corresponds to `Avocado Toast`, which we know has an `id` of 1223.
+
+##### Request:
+
+`DELETE http://www.myrestaurant.com/dishes/:id`
```

Comment:
if you are doing this on one line by itself you probably want triple backticks, here and everywhere else